src/kubelet/metric/pods.go

```diff
@@ -358,17 +358,28 @@ func replicasetNameToDeploymentName(rsName string) string {
 // which will be converted later to one metric per label.
 // It also prefix the labels with 'label.'
 func OneMetricPerLabel(rawLabels definition.FetchedValue) (definition.FetchedValue, error) {
-	labels, ok := rawLabels.(map[string]string)
-	if !ok {
-		return rawLabels, errors.New("error on creating kubelet label metrics")
-	}
+	return MapToPrefix("label.")(rawLabels)
+}
 
-	modified := make(definition.FetchedValues, len(labels))
-	for k, v := range labels {
-		modified[fmt.Sprintf("label.%v", k)] = v
```

Comment:
Seems this refactoring is not related to this commit, could be done in separate commit.